set.mm

```diff
@@ -382517,10 +382525,20 @@ proved using the inner form (see theorem 9.6 of [Schwabhauser] p. 69
 
   ${
     $d a b g i j $.
-    $( Define the relation congruence bewteen shapes.  Definition 4.4 of
-       [Schwabhauser] p. 35.  Ideally, we would define this for functions of
-       any set, but we will used words (functions over ` NN ` ) in most cases.
-       (Contributed by Thierry Arnoux, 3-Apr-2019.) $)
+    $( Define the relation congruence between shapes.  Definition 4.4 of
+       [Schwabhauser] p. 35.  Two shapes are congruent if all corresponding
+       segments between all corresponding points are congruent.  Ideally, we
+       would define this for functions of any set, but we will use words
+       (functions over ` NN ` ) in most cases.  Many books define triangle
+       congruence as requiring both the congruence of all corresponding
+       segments and of all corresponding angles.  However, ~ dfcgrg2 proves
+       that these two definitions are equivalent, so we'll use this definition
+       from [Schwabhauser] p. 35 instead because it's simpler.  Once two
+       triangles are proven congruent, you can use various theorems to prove
+       that corresponding parts of congruent triangles are congruent (CPCTC),
+       for example, see ~ cgr3simp1 , ~ cgr3simp2 , ~ cgr3simp3 , ~ cgrcgra ,
+       permutation laws such as ~ cgr3swap12 , and ~ dfcgrg2 .  (Contributed by
```

Comment:
I think this is not the place to talk about triangles and even less angles. This definition is about the congruence of general shapes. The two primitive notions in Tarski geometry are betweenness and congruence of segments. Congruence of angles is only a derived notion defined later in terms of it, so it makes no sense to define angle congruence using angles.

Reply:
I disagree. Many people are going to expect "shape congruence" to include angle congruence. This definition does *not* include the expected angle congruence, so we need to explain why not. A short justification, with links to relevant proofs, provides that justification.

Reply:
Are you sure you're not confusing with Hilbert geometry ?  Hilbert geometry contains angle congruence as a primitive notion, but Tarski geometry does not. These "many books" and "many people" you're talking about are probably developing geometry in other systems than the one used in set.mm. If that is indeed what you mean, then one could write something like:
> Other axiomatic systems of geometry, like Hilbert's system, also have a primitive notion of angle congruence, and therefore, congruence of triangles (which are simply "shapes" with three points) is sometimes defined in these systems using both (segment) congruence and angle congruence.  In Tarski's system, angle congruence is a derived notion, defined in df-.... and theorems ... prove that congruent triangles have congruent angles [and also all the SAS and CPTC stuff].

Reply:
Maybe the additional part of the comment (modified as @benjub proposed) should be moved to the section header.

Reply:
@benjub - I like the idea of your text explaining that the system we're using doesn't have angle congruence as a primitive notion. I'll take your text & modify it to try to make it "flow" in this discussion about definitions.

Even with that statement, though, we need to justify why we (and Tarski) are willing to define congruence of shapes without including congruence of angles. Many typical geometry materials, including those used worldwide in high school, include angles as a primitive fo defining the congruence of triangles. Oh, and don't say high school geometry is irrelevant; Tarski often taught high school geometry In the 1920s and 30s, and this led to his interest in geometry that we're formalizing. This looks like a big deviation, and we can better than an appeal to authority.

As evidence, here are some of sample materials I've found that *expressly* define congruence of triangles as requiring congruence of BOTH their corresponding angles AND their corresponding segments:

* "Two triangles are congruent if a one-to-one correspondence can be set up between their vertices so that corresponding sides are congruent and corresponding angles are congruent." - Euclidean and Non-Euclidean Geometries: Development and History by Marvin Jay Greenberg, 1974, page 69
* "The (example) two triangles are congruent because they are geometrically equivalent. All corresponding sides have equal lengths and all corresponding angles have equal measures." - Advanced Mathematics (Second Edition) by Saxon, Page 67.
* "Two triangles are congruent if their corresponding sides are equal in length, and their corresponding angles are equal in measure." - Wikipedia, [Congruence(geometry)](https://en.wikipedia.org/wiki/Congruence_(geometry)#:~:text=Congruence%20of%20triangles,-See%20also%3A%20Solution&text=Two%20triangles%20are%20congruent%20if,angles%20are%20equal%20in%20measure.)

It's *okay* that we don't include angles in the definition, there are even good reasons for it. We just need a brief comment explaining why why don't. Saying "that's what Tarski did" isn't enough, there's a good reason why Tarski did this that isn't merely an appeal to authority.

A brief note in the header as *well* would make sense, but people who surf this material though hyperlinks may end up on this definition of shape congruence without any context. Let's give them that context, and frankly, the details about shape congruence belong here, in the definition shape congruence.

Reply:
Thanks for the specific citations in these two books. I'll try to have a look because for now I do not know their axiom systems nor how they define "congruent angles".  By the way, I note that Greenberg indeed uses unordered shapes (i.e. finite sets as opposed to finite sequences), as I proposed in https://github.com/metamath/set.mm/pull/2994#discussion_r1083284350 before I saw that Schwabhauser uses sequences/tuples. I think the unordered shapes are more natural (a triangle does not care about the order in which you enumerate its vertices). But maybe that's too much worl for @tirix to modify it for a too little gain.

The "good reason why Tarski did it" is that angle congruence can be defined in terms of congruence and betweenness alone. So these two primitive notions suffice. This is connected to my question to @tirix in https://github.com/metamath/set.mm/pull/3028#discussion_r1096788692

As for the specific text, I'll let you @david-a-wheeler see, because it looks like you now have all the parts.

Reply:
I think the ordered sequence representation (words) chosen by @tirix is a good one. If triangle ABC is congruent to DEF, that does *NOT* mean that triangle BAC is necessarily congruent. So while in some sense the "order doesn't matter", when *referring* to a particular shape for making congruence statements, you typically *do* identify the shape by listing a specific order. I view this as a "shorthand for clearly identifying a specific shape".

Reply:
That's why the definition that Greenberg and I (in https://github.com/metamath/set.mm/pull/2994#discussion_r1083284350) proposed are formulated using a bijection. A triangle is defined by a set of three points, not a sequence of three points.

Reply:
> I'll try to have a look because for now I do not know their axiom systems nor how they define "congruent angles". 

The *usual* definition is, I believe, "congruent angles have equal measures".

In any case, I hope I've convinced you that there are *other* systems that regularly refer to "congruent angles" when they define "congruent shapes/triangles". All I want to do is provide a brief justification for the decision to *not* do that here, along with links to relevant proofs. It's not a change in the definition, but an explanation to help the reader understand why this specific definition is used.

Reply:
Then what is the "measure" of an angle ! This is actually trickier than one may think. (Actually, the "measure" of an angle is often defined to be its equivalence class for the angle congruence relation, so it's the other way around; it's similar to a "distance" being defined as an equivalence class for the segment congruence relation.)

I know there are other systems that have angle congruence as a primitive notion (I mentioned Hilbert's as such a system above); my point is that they are *other* systems. I agree with the explanation that you now wrote (my only problem with the first version was that the distinction between the different systems was not clear).

Reply:
@benjub :
> I agree with the explanation that you now wrote (my only problem with the first version was that the distinction between the different systems was not clear).

Excellent! Good to hear, I think the results are better than my first try for sure. Thank you! Given what you just said, I'm marking this as "resolved".